ethers-core/src/abi/mod.rs

```diff
@@ -100,12 +100,7 @@ impl<T: AbiArrayType, const N: usize> AbiType for [T; N] {
 
 impl<T: AbiArrayType, const N: usize> AbiArrayType for [T; N] {}
 
-impl<const N: usize> AbiType for [u8; N] {
-    fn param_type() -> ParamType {
-        ParamType::FixedBytes(N)
-    }
-}
-impl<const N: usize> AbiArrayType for [u8; N] {}
+impl AbiArrayType for u8 {}
```

Comment:
this defeats the whole purpose of this trait.

Reply:
The currently pushed code is certainly wrong. Can you check the comment https://github.com/gakonst/ethers-rs/pull/1140#issuecomment-1098345741 regarding the manual impls for u8 tuples? And for the boader issue and source of error https://github.com/gakonst/ethers-rs/pull/1140#issuecomment-1098363286